test/boilerplate_suggestors/advanced_props_and_state_class_migrator_test.dart

```diff
@@ -632,6 +632,61 @@ void main() {
               'FooProps': 'FooPropsMixin',
             });
           });
+
+          test('and is abstract', () {
+            converter.setConvertedClassNames({
+              'LayoutPropsMixin': 'LayoutPropsMixin',
+            });
+
+            testSuggestor(
+              expectedPatchCount: 8,
+              input: r'''
+              @AbstractProps()
+              abstract class _$AbstractBlockProps extends UiProps
+                  with
+                      LayoutPropsMixin,
+                      // ignore: mixin_of_non_class, undefined_class
+                      $LayoutPropsMixin,
+                      BlockPropsMixin,
+                      // ignore: mixin_of_non_class, undefined_class
+                      $BlockPropsMixin
+                  implements
+                      BlockClassHelperMapView {
+                String foo;
+                int bar;
+              }
+              
+              @AbstractComponent2()
+              abstract class AbstractBlockComponent<T extends AbstractBlockProps> extends UiComponent2<T>
+                  with LayoutMixin<T>, BlockMixin<T> {}
+            ''',
+              expectedOutput: r'''
+              @AbstractProps()
+              mixin AbstractBlockPropsMixin on UiProps implements BlockClassHelperMapView, LayoutPropsMixin, BlockPropsMixin, // ignore: mixin_of_non_class, undefined_class
+                      $BlockPropsMixin {
+                String foo;
+                int bar;
+              } 
+              
+              abstract class AbstractBlockProps = UiProps
```

Comment:
We should really never have a props class that's abstract, since we can't extend from it with the new boilerplate. In this case, I'd expect us to eliminate the abstract class.

This case may be difficult to handle entirely in codemod, so I'd be down to just emit a FIXME for this case

Reply:
Oh wait, is this case handled by another suggestor? If so, maybe we should test this more functionally with an AggregateSuggestor, since the intermediate states aren't as important

Reply:
Yes - the codemod used to remove `abstract` altogether... I'll just revert this change.

Reply:
nevermind - i will leave them abstract and make them interfaces as suggested here: https://github.com/Workiva/over_react_codemod/pull/81#discussion_r379032898